bdb/summarize.c

```diff
@@ -361,11 +361,9 @@ int bdb_summarize_table(bdb_state_type *bdb_state, int ixnum, int comp_pct,
         goto done;
     }
 
-#if defined(_IBM_SOURCE) || defined(__linux__)
+#if defined(POSIX_FADV_SEQUENTIAL)
     // inform kernel that we will be accessing file sequentially
-    // and that we won't need the file to be cached
-    fdatasync(fd);
-    posix_fadvise(fd, 0, 0, POSIX_FADV_DONTNEED | POSIX_FADV_SEQUENTIAL);
```

Comment:
`fadvise()` flags cannot be combined. Plus we probably do not want to use `POSIX_FADV_DONTNEED` for it'll clear pages from the index out of kernel page cache, which may be undesirable.

Reply:
The idea was to have fewest number of pages in kernel cache to be nice to the system--these pages are not needed at all once we have processed them. The fact that we can't combine the flags is indeed unfortunate.

Reply:
>> these pages are not needed at all once we have processed them

This isn't entirely true. And this is why `POSIX_FADV_DONTNEED` may not be appropriate here. Pages processed by analyze may be needed by the normal operations such as an index lookup or modification. Forcing these pages out therefore therefore isn't desirable.

Reply:
In fact the optimization here makes very little sense with DIRECTIO which bypasses kernel page cache :)